activesupport/test/cache/behaviors/connection_pool_behavior.rb

```diff
@@ -50,6 +50,35 @@ def test_no_connection_pool
     end
   end
 
+  def test_deprecated_connection_pool_works
+    Thread.report_on_exception, original_report_on_exception = false, Thread.report_on_exception
+
+    threads = []
+
+    emulating_latency do
+      assert_deprecated do
+        cache = ActiveSupport::Cache.lookup_store(*store, { pool_size: 2, pool_timeout: 1 }.merge(store_options))
+        cache.read("foo")
+
+        assert_raises Timeout::Error do
+          # One of the three threads will fail in 1 second because our pool size
+          # is only two.
+          3.times do
+            threads << Thread.new do
```

Comment:
I don't think you need such a complicated test, I also fear it will be flaky. 

I'd recommend to just check wether pooling is enabled, you can use `instance_variable_get` or `send` if necessary to check the internal state of the store.

Reply:
I just replicated the test above 😄 Updated with the suggestion.